db-queue-spring/src/main/java/ru/yoomoney/tech/dbqueue/spring/dao/H2QueuePickTaskDao.java

```diff
@@ -0,0 +1,219 @@
+package ru.yoomoney.tech.dbqueue.spring.dao;
+
+import org.springframework.jdbc.core.JdbcOperations;
+import ru.yoomoney.tech.dbqueue.api.TaskRecord;
+import ru.yoomoney.tech.dbqueue.config.QueueTableSchema;
+import ru.yoomoney.tech.dbqueue.dao.PickTaskSettings;
+import ru.yoomoney.tech.dbqueue.dao.QueuePickTaskDao;
+import ru.yoomoney.tech.dbqueue.settings.QueueLocation;
+import ru.yoomoney.tech.dbqueue.settings.TaskRetryType;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public class H2QueuePickTaskDao implements QueuePickTaskDao {
+    private final JdbcOperations jdbcTemplate;
+    private final QueueTableSchema queueTableSchema;
+    private final PickTaskSettings pickTaskSettings;
+
+    public H2QueuePickTaskDao(@Nonnull final JdbcOperations jdbcOperations,
+                              @Nonnull final QueueTableSchema queueTableSchema,
+                              @Nonnull final PickTaskSettings pickTaskSettings) {
+
+        this.jdbcTemplate = Objects.requireNonNull(jdbcOperations, "jdbc template can't be null");
+        this.queueTableSchema = Objects.requireNonNull(queueTableSchema, "table schema can't be null");
+        this.pickTaskSettings = Objects.requireNonNull(pickTaskSettings, "settings can't be null");
+
+        this.jdbcTemplate.update(
```

Comment:
Do we need a transaction here?

Reply:
Yes, you are right. Creating a stored procedure must be in a transaction.